core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -637,6 +637,16 @@ private[deploy] class Master(
       }
       freeWorkers = freeWorkers.filter(canLaunchExecutor)
     }
+
+    val numExecutorsScheduled = assignedExecutors.sum
+    val numExecutorsLaunched = app.executors.size
+    // Check to see if we managed to launch the requested number of executors
+    if(numUsable != 0 && numExecutorsLaunched != app.executorLimit &&
+      numExecutorsScheduled != app.executorLimit) {
```

Comment:
How are `numExecutorsLaunched` and `numExecutorsScheduled` related to each other? Also here we probably want to do an inequality check just in case.

Also style: need space after `if`